### ISSUE TITLE
Use declarative encoding for `http.middleware`

### ISSUE THREAD

**Issue by tusharmath** [zio/zio-http#701]
**What is not easy to do right now?**
Currently calling the `middleware` method on `http` runs the complete Http app using ZIO (executional encoding).
This is inconsistent with the rest of the design (declarative), it unnecessarily depends on ZIO and is not the most performant way of applying a middleware on an Http.

**Describe the solution you'd like**
Create a new type called `RunMiddleware` which extends `Http`, like —

```scala
RunMiddleware[R, E](app: HttpApp[R, E], mid: Middleware[R, E]) extends HttpApp[R, E]
``` 

**Why is it important?**
For consistency and performance.

**Additional context**
None

**Comment by brendo-m** [comment]
I can pick this one up

**Comment by brendo-m** [comment]
https://github.com/brendo-m/zio-http/commit/e6c6559c595144a113e9df807d148879b4034e98

**Comment by tusharmath** [comment]
@brendo-m With the recent changes in Middleware, it should now be possible. Would you like to give it a try?

**Comment by brendo-m** [comment]
Sure, this should do it: https://github.com/dream11/zio-http/pull/869

**Comment by girdharshubham** [comment]
Merged in #869

Thanks @brendo-m
